README.md
UMC for Wesnoth 1.14


Sharifs/CHANGELOG
###
#textdomain  wesnoth-Sharifs

			 CHANGELOG
________________________________________________________________________________

Sharifs is the name of an altered Dunefolk faction.
It delivers two more units: the bomber-engineer and the martial arts assassin (hashishi).
==== Credits & thanks go to the creators of ====
- Ageless_Era by Mnewton1,various (see credits there)
________________________________________________________________________________

version 1.8.4
    - release of of Wesnoth 1.14 RC3 version on Wesnoth 1.13.14 server

version 1.8.2a
    - new version scheme:
    - 1.8 represents the year 2018 and
    -  .2 stands for the month of that year, February
    - upgraded to Wesnoth's new naming of the Dunefolk-faction,
      formerly known as 'Khalifate' (A Caliphate is an islamic
      state under the leadership of a caliph, considered
      to be a religious successor to prophet Mu-ham-mad)

1.7.6x updates due to error-log errors
	+ introduction of lvl0 unit Felach

2.0.16.12.04c
	-base damage of bomber attack reduced from 30 to 24

2.0.16.3.15
	= minor fix

2.0.15.9
	= changed [unit_type] id-strings due to a bug:
		- bug renders saved games and saved replays 'invalid' or 'corrupted'
		- it was caused by the "-" in the id-strings like id=tiv-naffat
		- changed to "_", e.g. id=tiv_naffat
	= minor bug fixes

Version 2.0.15.8
	= version release as of August 2015
	= verry slightly lessened the base values of the Khalifate units
		via base unit include and only HP XP attack values altered
	= changed all units to alignment=liminal (as the factions feature), but not falcon

Version 1.0.3
	= minor bug fix

Version 1.0.2
 = tested stable version
	+ small adaptions in code style
	+ ad of Desert Map-Pack (as a dependency)

Version 1.0.1 to 1.0.2
	= unpublished testing versions

Version 0.3.2
 	= small adaptions

Version 0.3.0
 	= initial release

________________________________________________________________________________
Initially released end of June, 2015, on Wesnoth 1.12.2 server.


Tunnel_Rats/part05/lua/theme.lua
-- #textdomain wesnoth-Tunnel_Rats_part05

local _ = wesnoth.textdomain "wesnoth-Tunnel_Rats_part05"
local old_unit_status = wesnoth.theme_items.unit_status

function wesnoth.theme_items.unit_status()
	local u = wesnoth.get_displayed_unit()
	if not u then return {} end
	local s = old_unit_status()

	if u.status.dehydrated then
		table.insert(s, { "element", {
			image = "misc/dehydration-status.png",
			tooltip = _ "dehydrated: This unit is dehydrated. It will lose 4 HP and have its damage reduced by 1 each turn during the day unless prevented by healers or cured by water at an oasis.\n\nUnits cannot be killed or deal no damage as a result of dehydration."
		} })
	end

	if u.status.stunned then
		table.insert(s, { "element", { image = "misc/stunned-status-icon.png",
			tooltip = _ "stunned: This unit is stunned. It cannot enforce its Zone of Control."
		} } )
	end

	if u.status.dazed then
		table.insert(s, { "element", { image = "misc/dazed-status-icon.png",
			tooltip = _ "dazed: This unit is dazed. It suffers a -10% penalty to both its defense and chance to hit (except for magical attacks)."
		} } )
	end

	return s
end


Tunnel_Rats/part00/lua/gender_selection00.lua
-- #textdomain wesnoth-Tunnel_Rats_part00

local helper = wesnoth.require "helper"
local T = helper.set_wml_tag_metatable {}
local wml_actions = wesnoth.wml_actions
local _ = wesnoth.textdomain "wesnoth-Tunnel_Rats_part00"

function wml_actions.select_gender()
	local gender_selection_dialog = {
		maximum_height = 250,
		maximum_width = 400,
		T.helptip { id="tooltip_large" }, -- mandatory field
		T.tooltip { id="tooltip_large" }, -- mandatory field
		T.grid {
			T.row {
				T.column {
					grow_factor = 1,
					border = "all",
					border_size = 5,
					horizontal_alignment = "left",
					T.label {
						definition = "title",
						label = _"Select the Tutorial's Main Character"
					}
				}
			},
			T.row {
				T.column {
					grow_factor = 1,
					border = "all",
					border_size = 5,
					horizontal_alignment = "left",
					T.label {
						label = _"Who do you want to play?"
					}
				}
			},
			T.row {
				T.column {
					T.grid {
						T.row {
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.image {
									label = "portraits/maincharacters/maincharacter_female.png"
								}
							},
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.image {
									label = "portraits/maincharacters/maincharacter_male.png~TC(1,magenta)"
								}
							}

						},
						T.row {
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.button {
									label = _"Shieldmaiden",
									return_value = 2
								}
							},
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.button {
									label = _"Watchman",
									return_value = 1
								}
							}

						}
					}
				}
			}
		}
	}

	local gender = wesnoth.show_dialog(gender_selection_dialog)
	local unit = wesnoth.get_variable("gender_store")

	if gender == 2 then
		wesnoth.put_unit({
			type = "TUNRA_protagonistine",
            gender = "female",
			id = unit.id,
			name = _"Hermine",
			unrenamable = true,
			profile="portraits/maincharacter_female.png",
			canrecruit = true,
            extra_recruit = "",
			facing = unit.facing,
		}, unit.x, unit.y )
	else
		wesnoth.put_unit({
			type = "TUNRA_protagonist",
            gender = "male",
			id = unit.id,
			name = _"Herman",
			unrenamable = true,
			profile="portraits/maincharacter_male.png",
			canrecruit = true,
            extra_recruit = "",
			facing = unit.facing,
		}, unit.x, unit.y )
	end
	wesnoth.redraw {}
end


Tunnel_Rats/part01/lua/gender_selection01.lua
-- #textdomain wesnoth-Tunnel_Rats_part01

local helper = wesnoth.require "helper"
local T = helper.set_wml_tag_metatable {}
local wml_actions = wesnoth.wml_actions
local _ = wesnoth.textdomain "wesnoth-Tunnel_Rats_part01"

function wml_actions.select_gender()
	local gender_selection_dialog = {
		maximum_height = 250,
		maximum_width = 400,
		T.helptip { id="tooltip_large" }, -- mandatory field
		T.tooltip { id="tooltip_large" }, -- mandatory field
		T.grid {
			T.row {
				T.column {
					grow_factor = 1,
					border = "all",
					border_size = 5,
					horizontal_alignment = "left",
					T.label {
						definition = "title",
						label = _"Select the Epyllion's Main Character"
					}
				}
			},
			T.row {
				T.column {
					grow_factor = 1,
					border = "all",
					border_size = 5,
					horizontal_alignment = "left",
					T.label {
						label = _"Who do you want to play?"
					}
				}
			},
			T.row {
				T.column {
					T.grid {
						T.row {
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.image {
									label = "portraits/maincharacters/maincharacter_female.png"
								}
							},
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.image {
									label = "portraits/maincharacters/maincharacter_male.png~TC(1,magenta)"
								}
							}

						},
						T.row {
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.button {
									label = _"Shieldmaiden",
									return_value = 2
								}
							},
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.button {
									label = _"Watchman",
									return_value = 1
								}
							}

						}
					}
				}
			}
		}
	}

	local gender = wesnoth.show_dialog(gender_selection_dialog)
	local unit = wesnoth.get_variable("gender_store")

	if gender == 2 then
		wesnoth.put_unit({
			type = "TUNRA_protagonistine",
            gender = "female",
			id = unit.id,
			name = _"Hermine",
			unrenamable = true,
			profile="portraits/maincharacter_female.png",
			canrecruit = true,
            extra_recruit = "",
			facing = unit.facing,
		}, unit.x, unit.y )
	else
		wesnoth.put_unit({
			type = "TUNRA_protagonist",
            gender = "male",
			id = unit.id,
			name = _"Herman",
			unrenamable = true,
			profile="portraits/maincharacter_male.png",
			canrecruit = true,
            extra_recruit = "",
			facing = unit.facing,
		}, unit.x, unit.y )
	end
	wesnoth.redraw {}
end


Tunnel_Rats/part02/lua/gender_selection02.lua
-- #textdomain wesnoth-Tunnel_Rats_part02

local helper = wesnoth.require "helper"
local T = helper.set_wml_tag_metatable {}
local wml_actions = wesnoth.wml_actions
local _ = wesnoth.textdomain "wesnoth-Tunnel_Rats_part02"

function wml_actions.select_gender()
	local gender_selection_dialog = {
		maximum_height = 250,
		maximum_width = 400,
		T.helptip { id="tooltip_large" }, -- mandatory field
		T.tooltip { id="tooltip_large" }, -- mandatory field
		T.grid {
			T.row {
				T.column {
					grow_factor = 1,
					border = "all",
					border_size = 5,
					horizontal_alignment = "left",
					T.label {
						definition = "title",
						label = _"Select the Epyllion's Main Character"
					}
				}
			},
			T.row {
				T.column {
					grow_factor = 1,
					border = "all",
					border_size = 5,
					horizontal_alignment = "left",
					T.label {
						label = _"Who do you want to play?"
					}
				}
			},
			T.row {
				T.column {
					T.grid {
						T.row {
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.image {
									label = "portraits/maincharacters/maincharacter_female.png"
								}
							},
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.image {
									label = "portraits/maincharacters/maincharacter_male.png~TC(1,magenta)"
								}
							}

						},
						T.row {
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.button {
									label = _"Shieldmaiden",
									return_value = 2
								}
							},
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.button {
									label = _"Watchman",
									return_value = 1
								}
							}

						}
					}
				}
			}
		}
	}

	local gender = wesnoth.show_dialog(gender_selection_dialog)
	local unit = wesnoth.get_variable("gender_store")

	if gender == 2 then
		wesnoth.put_unit({
			type = "TUNRA_protagonistine",
            gender = "female",
			id = unit.id,
			name = _"Hermine",
			unrenamable = true,
			profile="portraits/maincharacter_female.png",
			canrecruit = true,
            extra_recruit = "",
			facing = unit.facing,
		}, unit.x, unit.y )
	else
		wesnoth.put_unit({
			type = "TUNRA_protagonist",
            gender = "male",
			id = unit.id,
			name = _"Herman",
			unrenamable = true,
			profile="portraits/maincharacter_male.png",
			canrecruit = true,
            extra_recruit = "",
			facing = unit.facing,
		}, unit.x, unit.y )
	end
	wesnoth.redraw {}
end


Tunnel_Rats/part03/lua/gender_selection03.lua
-- #textdomain wesnoth-Tunnel_Rats_part03

local helper = wesnoth.require "helper"
local T = helper.set_wml_tag_metatable {}
local wml_actions = wesnoth.wml_actions
local _ = wesnoth.textdomain "wesnoth-Tunnel_Rats_part00"

function wml_actions.select_gender()
	local gender_selection_dialog = {
		maximum_height = 250,
		maximum_width = 400,
		T.helptip { id="tooltip_large" }, -- mandatory field
		T.tooltip { id="tooltip_large" }, -- mandatory field
		T.grid {
			T.row {
				T.column {
					grow_factor = 1,
					border = "all",
					border_size = 5,
					horizontal_alignment = "left",
					T.label {
						definition = "title",
						label = _"Select the Epyllion's Main Character"
					}
				}
			},
			T.row {
				T.column {
					grow_factor = 1,
					border = "all",
					border_size = 5,
					horizontal_alignment = "left",
					T.label {
						label = _"Who do you want to play?"
					}
				}
			},
			T.row {
				T.column {
					T.grid {
						T.row {
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.image {
									label = "portraits/maincharacters/maincharacter_female.png"
								}
							},
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.image {
									label = "portraits/maincharacters/maincharacter_male.png~TC(1,magenta)"
								}
							}

						},
						T.row {
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.button {
									label = _"Shieldmaiden",
									return_value = 2
								}
							},
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.button {
									label = _"Watchman",
									return_value = 1
								}
							}

						}
					}
				}
			}
		}
	}

	local gender = wesnoth.show_dialog(gender_selection_dialog)
	local unit = wesnoth.get_variable("gender_store")

	if gender == 2 then
		wesnoth.put_unit({
			type = "TUNRA_protagonistine",
            gender = "female",
			id = unit.id,
			name = _"Hermine",
			unrenamable = true,
			profile="portraits/maincharacter_female.png",
			canrecruit = true,
            extra_recruit = "",
			facing = unit.facing,
		}, unit.x, unit.y )
	else
		wesnoth.put_unit({
			type = "TUNRA_protagonist",
            gender = "male",
			id = unit.id,
			name = _"Herman",
			unrenamable = true,
			profile="portraits/maincharacter_male.png",
			canrecruit = true,
            extra_recruit = "",
			facing = unit.facing,
		}, unit.x, unit.y )
	end
	wesnoth.redraw {}
end


Tunnel_Rats/part05/lua/gender_selection05.lua
-- #textdomain wesnoth-Tunnel_Rats_part05

local helper = wesnoth.require "helper"
local T = helper.set_wml_tag_metatable {}
local wml_actions = wesnoth.wml_actions
local _ = wesnoth.textdomain "wesnoth-Tunnel_Rats_part00"

function wml_actions.select_gender()
	local gender_selection_dialog = {
		maximum_height = 250,
		maximum_width = 400,
		T.helptip { id="tooltip_large" }, -- mandatory field
		T.tooltip { id="tooltip_large" }, -- mandatory field
		T.grid {
			T.row {
				T.column {
					grow_factor = 1,
					border = "all",
					border_size = 5,
					horizontal_alignment = "left",
					T.label {
						definition = "title",
						label = _"Select the Epyllion's Main Character"
					}
				}
			},
			T.row {
				T.column {
					grow_factor = 1,
					border = "all",
					border_size = 5,
					horizontal_alignment = "left",
					T.label {
						label = _"Who do you want to play?"
					}
				}
			},
			T.row {
				T.column {
					T.grid {
						T.row {
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.image {
									label = "portraits/maincharacters/maincharacter_female.png"
								}
							},
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.image {
									label = "portraits/maincharacters/maincharacter_male.png~TC(1,magenta)"
								}
							}

						},
						T.row {
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.button {
									label = _"Shieldmaiden",
									return_value = 2
								}
							},
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.button {
									label = _"Watchman",
									return_value = 1
								}
							}

						}
					}
				}
			}
		}
	}

	local gender = wesnoth.show_dialog(gender_selection_dialog)
	local unit = wesnoth.get_variable("gender_store")

	if gender == 2 then
		wesnoth.put_unit({
			type = "TUNRA_heroine",
            gender = "female",
			id = unit.id,
			name = _"Hermine",
			unrenamable = true,
			profile="portraits/maincharacter_female.png",
			canrecruit = true,
            extra_recruit = "",
			facing = unit.facing,
		}, unit.x, unit.y )
	else
		wesnoth.put_unit({
			type = "TUNRA_hero",
            gender = "male",
			id = unit.id,
			name = _"Herman",
			unrenamable = true,
			profile="portraits/maincharacter_male.png",
			canrecruit = true,
            extra_recruit = "",
			facing = unit.facing,
		}, unit.x, unit.y )
	end
	wesnoth.redraw {}
end


Tunnel_Rats/part04/lua/gender_selection04.lua
-- #textdomain wesnoth-Tunnel_Rats_part04

local helper = wesnoth.require "helper"
local T = helper.set_wml_tag_metatable {}
local wml_actions = wesnoth.wml_actions
local _ = wesnoth.textdomain "wesnoth-Tunnel_Rats_part00"

function wml_actions.select_gender()
	local gender_selection_dialog = {
		maximum_height = 250,
		maximum_width = 400,
		T.helptip { id="tooltip_large" }, -- mandatory field
		T.tooltip { id="tooltip_large" }, -- mandatory field
		T.grid {
			T.row {
				T.column {
					grow_factor = 1,
					border = "all",
					border_size = 5,
					horizontal_alignment = "left",
					T.label {
						definition = "title",
						label = _"Select the Epyllion's Main Character"
					}
				}
			},
			T.row {
				T.column {
					grow_factor = 1,
					border = "all",
					border_size = 5,
					horizontal_alignment = "left",
					T.label {
						label = _"Who do you want to play?"
					}
				}
			},
			T.row {
				T.column {
					T.grid {
						T.row {
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.image {
									label = "portraits/maincharacters/maincharacter_female.png"
								}
							},
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.image {
									label = "portraits/maincharacters/maincharacter_male.png~TC(1,magenta)"
								}
							}

						},
						T.row {
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.button {
									label = _"Shieldmaiden",
									return_value = 2
								}
							},
							T.column {
								grow_factor = 1,
								border = "all",
								border_size = 5,
								T.button {
									label = _"Watchman",
									return_value = 1
								}
							}

						}
					}
				}
			}
		}
	}

	local gender = wesnoth.show_dialog(gender_selection_dialog)
	local unit = wesnoth.get_variable("gender_store")

	if gender == 2 then
		wesnoth.put_unit({
			type = "TUNRA_protagonistine",
            gender = "female",
			id = unit.id,
			name = _"Hermine",
			unrenamable = true,
			profile="portraits/maincharacter_female.png",
			canrecruit = true,
            extra_recruit = "",
			facing = unit.facing,
		}, unit.x, unit.y )
	else
		wesnoth.put_unit({
			type = "TUNRA_protagonist",
            gender = "male",
			id = unit.id,
			name = _"Herman",
			unrenamable = true,
			profile="portraits/maincharacter_male.png",
			canrecruit = true,
            extra_recruit = "",
			facing = unit.facing,
		}, unit.x, unit.y )
	end
	wesnoth.redraw {}
end
